sdk/core/Azure.Core/tests/OperationInternalOfTTests.cs

```diff
@@ -205,6 +218,35 @@ public async Task UpdateStatusPassesTheCancellationTokenToUpdateState([Values(tr
             Assert.AreEqual(originalToken, passedToken);
         }
 
+        [Test]
+        public async Task WaitForCompletionResponseCreatesDiagnosticScope([Values(true, false)] bool async, [Values(null, "CustomTypeName")] string operationTypeName)
+        {
+            using ClientDiagnosticListener testListener = new(DiagnosticNamespace);
+
+            string expectedTypeName = operationTypeName ?? nameof(TestOperation);
+            KeyValuePair<string, string>[] expectedAttributes = { new("key1", "value1"), new("key2", "value2") };
+            var operationInternal = new OperationInternal<int>(ClientDiagnostics, TestOperation.Succeeded(1), InitialResponse, operationTypeName, expectedAttributes);
+
+            _ = async
+                ? await operationInternal.WaitForCompletionResponseAsync(CancellationToken.None)
+                : operationInternal.WaitForCompletionResponse(CancellationToken.None);
+
+            testListener.AssertScope($"{expectedTypeName}.WaitForCompletionResponse", expectedAttributes);
```

Comment:
We should add a test for `WaitForCompletion` for Operation\<T\> as well.  (Operation does not have this method so doesn't need this test.)

Reply:
`WaitForCompletionResponse` is a public method could be called by operation. Why we don't need this?

Reply:
> WaitForCompletionResponse is a public method could be called by operation. Why we don't need this?

I am not saying we don't need the test for WaitForCompletionResponse.  I am saying that we need to **add** a test for WaitForCompletion to the OperationInternal\<T\> tests.  OperationInternal\<T\> does have a WaitForCompletion method.  I would like for us to add a test that shows that when `WaitForCompletion` is called, the scope name is `WaitForCompletion` and not `WaitForCompletionResponse`.